pkg/operations/kubernetes162upgradeagentnode.go

```diff
@@ -1,25 +1,74 @@
 package operations
 
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/Azure/acs-engine/pkg/api"
+	"github.com/Azure/acs-engine/pkg/armhelpers"
+
+	log "github.com/Sirupsen/logrus"
+)
+
 // Compiler to verify QueueMessageProcessor implements OperationsProcessor
 var _ UpgradeNode = &UpgradeAgentNode{}
 
 // UpgradeAgentNode upgrades a Kubernetes 1.5.3 agent node to 1.6.2
 type UpgradeAgentNode struct {
+	TemplateMap             map[string]interface{}
+	ParametersMap           map[string]interface{}
+	UpgradeContainerService *api.ContainerService
+	ResourceGroup           string
+	Client                  armhelpers.ACSEngineClient
 }
 
 // DeleteNode takes state/resources of the master/agent node from ListNodeResources
 // backs up/preserves state as needed by a specific version of Kubernetes and then deletes
 // the node
-func (kmn *UpgradeAgentNode) DeleteNode(*string) error {
+func (kan *UpgradeAgentNode) DeleteNode(vmName *string) error {
+	if err := CleanDeleteVirtualMachine(kan.Client, kan.ResourceGroup, *vmName); err != nil {
+		log.Fatalln(err)
+		return err
+	}
+
 	return nil
 }
 
 // CreateNode creates a new master/agent node with the targeted version of Kubernetes
-func (kmn *UpgradeAgentNode) CreateNode(int) error {
+func (kan *UpgradeAgentNode) CreateNode(poolName string, agentNo int) error {
+	poolCountParameter := kan.ParametersMap[poolName+"Count"].(map[string]interface{})
+	poolCountParameter["value"] = agentNo + 1
+	agentCount, _ := poolCountParameter["value"]
+	log.Infoln(fmt.Sprintf("Agent pool: %s, set count to: %d temporarily during upgrade...", poolName, agentCount))
+
+	poolOffsetVarName := poolName + "Offset"
+	templateVariables := kan.TemplateMap["variables"].(map[string]interface{})
+	templateVariables[poolOffsetVarName] = agentNo
+	agentOffset, _ := templateVariables[poolOffsetVarName]
+	log.Infoln(fmt.Sprintf("Agent offset: %v", agentOffset))
+
+	WriteTemplate(kan.UpgradeContainerService, kan.TemplateMap, kan.ParametersMap)
```

Comment:
Is this mostly for debug/troubleshooting? Since it will get written over many times during a full cluster upgrade?